libraries/chain/include/eosio/chain/webassembly/interface.hpp

```diff
@@ -100,175 +364,1547 @@ namespace webassembly {
          double _eosio_ui32_to_f64(uint32_t) const;
          double _eosio_ui64_to_f64(uint64_t) const;
 
-         // producer api
-         int32_t get_active_producers(legacy_span<account_name>) const;
-
-         // crypto api
-         void assert_recover_key(legacy_ptr<const fc::sha256>, legacy_span<const char>, legacy_span<const char>) const;
-         int32_t recover_key(legacy_ptr<const fc::sha256>, legacy_span<const char>, legacy_span<char>) const;
-         void assert_sha256(legacy_span<const char>, legacy_ptr<const fc::sha256>) const;
-         void assert_sha1(legacy_span<const char>, legacy_ptr<const fc::sha1>) const;
-         void assert_sha512(legacy_span<const char>, legacy_ptr<const fc::sha512>) const;
-         void assert_ripemd160(legacy_span<const char>, legacy_ptr<const fc::ripemd160>) const;
-         void sha256(legacy_span<const char>, legacy_ptr<fc::sha256>) const;
-         void sha1(legacy_span<const char>, legacy_ptr<fc::sha1>) const;
-         void sha512(legacy_span<const char>, legacy_ptr<fc::sha512>) const;
-         void ripemd160(legacy_span<const char>, legacy_ptr<fc::ripemd160>) const;
-
-         // permission api
-         bool check_transaction_authorization(legacy_span<const char>, legacy_span<const char>, legacy_span<const char>) const;
-         bool check_permission_authorization(account_name, permission_name, legacy_span<const char>, legacy_span<const char>, uint64_t) const;
-         int64_t get_permission_last_used(account_name, permission_name) const;
-         int64_t get_account_creation_time(account_name) const;
-
-         // authorization api
-         void require_auth(account_name) const;
-         void require_auth2(account_name, permission_name) const;
-         bool has_auth(account_name) const;
-         void require_recipient(account_name);
-         bool is_account(account_name) const;
-
-         // system api
+         /**
+          * Get the list of active producer names.
+          *
+          * @ingroup producer
+          * @param[out] producers - output buffer containing the names of the current active producer names.
+          *
+          * @return number of bytes required (if the buffer is empty), or the number of bytes written to the buffer.
+         */
+         int32_t get_active_producers(legacy_span<account_name> producers) const;
+
+         /**
+          * Tests a given public key with the recovered public key from digest and signature.
```

Comment:
It may be worth noting somehow that this assertion does not require the EOSIO canonical check to pass. That is, a signature that would have been disallowed during transaction acceptance would potentially be allowed here. Many contract users probably don't care about that detail.. but it's an important subtle behavior of the interface

Reply:
I am somewhat out of my element here, but if you have a suggested text i can add it, let me know what text should i add.